### PR TITLE
add the ability to add custom DataFetcher for certain fields

### PR DESCRIPTION
**Pull request by andimarek**
this gives the user of Nadel the ability to inject basically custom code for DataFetchers which is not expressed in the DSL

### CODE REVIEW

**Comment by andimarek** [comment]
the test is actually green: see https://travis-ci.org/graphql-java/nadel/builds/424219774

File: src/main/java/graphql/nadel/Nadel.java [hunk 1]
```diff
@@ -40,11 +42,22 @@
     private final Map<String, TypeDefinitionRegistry> typesByService = new LinkedHashMap<>();
     private final Map<String, SchemaNamespace> namespaceByService = new LinkedHashMap<>();
 
+    public Nadel(String dsl,
```
Comment (tsroka) [inline src/main/java/graphql/nadel/Nadel.java]: we need javadoc here. Or maybe make constructor private and use Builder and document each of the fields there?

File: src/test/groovy/graphql/nadel/NadelTest.groovy [hunk 2]
```diff
@@ -107,4 +115,54 @@ class NadelTest extends Specification {
...
+
+        TypeDefinitionRegistry typeDefinitionRegistry = new TypeDefinitionRegistry();
+
+
+        def fieldDefinition = FieldDefinition
+                .newFieldDefinition()
+                .name("specialField")
+                .type(new TypeName("String"))
+                .build()
+        ObjectTypeDefinition objectTypeDefinition = ObjectTypeDefinition.newObjectTypeDefinition()
```
Comment (tsroka) [inline src/test/groovy/graphql/nadel/NadelTest.groovy]: I am thinking that we can do better: field definition and object type definition should be built from DSL, not manually only dataFetcher should be supplied. We just need something to tell nadel that this fetcher should be used, how about:
`type Foo {
specialField: String <= $innerFetcher.specialFieldFetcher
}`
and then we could just do with only providing fetchers  -> no need to build types again etc
what do you think?

**Comment by andimarek** [comment]
will not be merged

**Pull request closed without merging**